### PR TITLE
Clean up curses interface rendering

### PR DESCRIPTION
**Pull request by ericaltendorf**
Main functional change: switch from `stdscr.getmaxyx()` to `os.popen('stty size', 'r')`, since the former doesn't seem to work (at least on my system).  Possibly a bug in curses, or a bad compile-time setting for the python curses library.  Some background: https://stackoverflow.com/questions/33906183#33906270

Allow "window" (curses window, more like pane or tile) sizing to be dynamic based on terminal resizing and based on the size of the `archive_report`, which can change dynamically during run time.

And, move a bunch of code around to be a little better structured.

### CODE REVIEW

**Review by altendky — CHANGES_REQUESTED** [review]
> And, move a bunch of code around to be a little better structured.

Definitely helpful to do that separately when it comes to others looking at the change(s).  As is the diff shows just that and it takes awhile to figure out if there are functional changes in the moved code.

Functionally I do see a difference.  Admittedly I'm running over mosh right now so...  extra whatever funny business there.  Both original and this respond to size changes.  I should probably run it locally.  You might be interested in seeing it over mosh (before and after).  I don't know what is desired so reaching conclusions when they both look a bit odd is tough.

I didn't really find any problems.  I can't say I went through it all line by line somewhat due to the mixed-purpose PR as mentioned above.  There's active development so problems will presumably get fixed if missed here.

I will note that this does shift towards being more Linux specific I think.  Not that I know how well curses works across platforms, but someone didn't mention today that they had it running in Windows.  So...  you may or may not want to make the `stty` use conditional either now or later.

File: interactive.py [hunk 1]
```diff
@@ -153,41 +114,118 @@ def curses_main(stdscr):
...
+                        else:
+                            cmd = status_or_cmd
+                            log.log('Starting archive: ' + cmd)
+
+                            # TODO: do something useful with output instead of DEVNULL
+                            p = subprocess.Popen(cmd,
+                                    shell=True,
+                                    stdout=subprocess.DEVNULL,
+                                    stderr=subprocess.STDOUT,
+                                    start_new_session=True)
```
[the remark this answers is not in the document]
Comment (ericaltendorf) [reply interactive.py:146]: sorry about that.  more narrowly targeted changes in the future will hopefully reduce this kind of confusion.

File: interactive.py [hunk 2]
```diff
@@ -109,35 +85,20 @@ def curses_main(stdscr):
...
-                        'terminal window.')
+    # Create windows.  We'll size them in the main loop when we have their content.
+    header_win = curses.newwin(1, 1, 1, 0)
+    log_win = curses.newwin(1, 1, 1, 0)
+    jobs_win = curses.newwin(1, 1, 1, 0)
+    dirs_win = curses.newwin(1, 1, 1, 0)
 
     jobs = Job.get_running_jobs(dir_cfg['log'])
-    last_refresh = datetime.datetime.now()
+    last_refresh = datetime.datetime.utcfromtimestamp(0)  # Beginning of time
```
[the remark this answers is not in the document]
Comment (ericaltendorf) [reply interactive.py]: Thanks, fixed.

**Comment by ericaltendorf** [comment]
Fixed the utcfromtimestamp(0) cruft, and made use of `stty size` conditional on a config parameter (default off, since it seems others are not experiencing the screen-resize issues I was experiencing).

**Review by altendky — APPROVED** [review]

**Pull request merged**